cli/cli/main.go

```diff
@@ -40,8 +49,59 @@ func main() {
 	})
 
 	if err := commands.RootCmd.Execute(); err != nil {
-		// We don't actually need to print the error because Cobra will do it for us
+		maybeCleanedError := getErrorMessageToBeDisplayedOnCli(err)
+		errorMessageFromCli := maybeCleanedError.Error()
+		// cobra uses this method underneath as well to print errors
+		// so just used it directly.
+		commands.RootCmd.PrintErrln(errorPrefix, errorMessageFromCli)
+
+		// if unknown command is entered - display help command
+		if strings.Contains(errorMessageFromCli, commandNotFound) {
+			commands.RootCmd.PrintErrf("Run '%v --help' for usage.\n", commands.RootCmd.CommandPath())
+		}
 		os.Exit(errorExitCode)
 	}
 	os.Exit(successExitCode)
 }
+
+func getErrorMessageToBeDisplayedOnCli(errorWithStacktrace error) error {
+	// if we are running in the debug mode, just return the error with stack-traces back to the client
+	if logrus.GetLevel() == logrus.DebugLevel {
+		return errorWithStacktrace
+	}
+
+	// silently catch the file logger error and print it in the debug mode
+	// users should not worry about this error
+	// downside is that we may lose stack-traces during file logger failures
+	fileLogger, err := out.GetFileLogger()
+	if err != nil {
+		logrus.Debugf("Error occurred while getting the file logger %+v", err)
```

Comment:
IMO silenting an error is always bad, I'd bias toward being loud here. At least WARN, if not ERROR. Hopefully that should never happen, so we don't run the risk of being too noisy. But when that happens, at least people know their logs might be missing stuffs.